cg/meta/demultiplex/status_db_storage_functions.py

```diff
@@ -75,10 +70,96 @@ def store_sequencing_metrics_in_status_db(flow_cell: FlowCellDirectoryData, stor
+        SampleLaneSequencingMetrics
+    ] = create_undetermined_sequencing_metrics_for_flow_cell(
+        flow_cell_directory=flow_cell.path,
+        bcl_converter=flow_cell.bcl_converter,
+        non_pooled_lanes_and_samples=non_pooled_lanes_and_samples,
+    )
+
+    for undetermined_metric in metrics:
+        sample_id: str = undetermined_metric.sample_internal_id
+        lane: int = undetermined_metric.flow_cell_lane_number
+
+        existing_metric: SampleLaneSequencingMetrics = (
+            store.get_metrics_entry_by_flow_cell_name_sample_internal_id_and_lane(
+                flow_cell_name=flow_cell.id, sample_internal_id=sample_id, lane=lane
+            )
+        )
+        if existing_metric:
+            update_existing_metric(existing_metric=existing_metric, new_metric=undetermined_metric)
+        else:
+            store.session.add(undetermined_metric)
+    store.session.commit()
+
+
+def update_existing_metric(
+    existing_metric: SampleLaneSequencingMetrics, new_metric: SampleLaneSequencingMetrics
+) -> None:
+    """Update an existing metric with a new metric."""
+
+    combined_q30_percentage: float = combine_q30_percentage(
+        existing_metric=existing_metric, new_metric=new_metric
+    )
+    combined_mean_quality_score: float = combine_mean_quality_score(
+        existing_metric=existing_metric, new_metric=new_metric
+    )
+    combined_reads: int = combine_reads(existing_metric=existing_metric, new_metric=new_metric)
+
+    existing_metric.sample_base_percentage_passing_q30 = combined_q30_percentage
+    existing_metric.sample_base_mean_quality_score = combined_mean_quality_score
+    existing_metric.sample_total_reads_in_lane = combined_reads
+
+
+def combine_mean_quality_score(
+    existing_metric: SampleLaneSequencingMetrics, new_metric: SampleLaneSequencingMetrics
+) -> float:
+    """Calculate the weighted average of two mean quality scores."""
+    return weighted_average(
+        total_1=existing_metric.sample_total_reads_in_lane,
+        percentage_1=existing_metric.sample_base_mean_quality_score,
+        total_2=new_metric.sample_total_reads_in_lane,
+        percentage_2=new_metric.sample_base_mean_quality_score,
+    )
+
+
+def combine_q30_percentage(
+    existing_metric: SampleLaneSequencingMetrics, new_metric: SampleLaneSequencingMetrics
+) -> float:
+    """Calculate the weighted average of two q30 percentages."""
+    return weighted_average(
+        total_1=existing_metric.sample_total_reads_in_lane,
+        percentage_1=existing_metric.sample_base_percentage_passing_q30,
+        total_2=new_metric.sample_total_reads_in_lane,
+        percentage_2=new_metric.sample_base_percentage_passing_q30,
+    )
+
+
+def combine_reads(
+    existing_metric: SampleLaneSequencingMetrics, new_metric: SampleLaneSequencingMetrics
+) -> int:
+    """Combine the total reads from two metrics."""
+    return existing_metric.sample_total_reads_in_lane + new_metric.sample_total_reads_in_lane
```

Comment:
I personally don't think this needs its own function.

Reply:
I think it is essential 😠